components/concept/storage/src/main/java/mozilla/components/concept/storage/CleanableStorage.kt

```diff
@@ -0,0 +1,25 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.concept.storage
+
+/**
+ * Storage that allows to stop and clean in progress operations.
+ */
+interface CleanableStorage {
```

Comment:
Extracted this from `Storage` as a separate interface as `HistoryMetadataStorage` needs to support the new `cleanup..` methods but don't have to support [the other two methods from `Storage`](https://github.com/mozilla-mobile/android-components/blob/6c0fe91ab83614265f533b819bf406fc046b2227/components/concept/storage/src/main/java/mozilla/components/concept/storage/Storage.kt#L11-L19).